lib/ruote/postgres/storage.rb

```diff
@@ -297,14 +298,30 @@ def has_json?
         server_version[0] >= 9 && server_version[1] >= 2
       end
 
+      def reconnect
+        @pg = db_connect
+      end
+
+      def db_connect
+        PGconn.connect_start(@pg.host, @pg.port, @pg.options, @pg.tty, @pg.db, @pg.user, @pg.pass)
+      end
+
       def safe_pg(&block)
+        retries ||= 0
+
         @mutex.synchronize do
           yield
         end
       rescue *CONNECTION_ERRORS => e
+        if @retry_on_connection_error && retries < 1
```

Comment:
Interesting but In my view it's not needed. The kind of error we are facing it's a network issue that closes the PG connection. With 1 retry it reconnects and it works, if that's not the case it may well be another issue and we just raise the error.